applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing_matrix_free.h

```diff
@@ -142,42 +142,51 @@ class MapperVertexMorphingMatrixFree : public Mapper
         mValuesDestination[2].clear();
 
         // Perform mapping
-        for(auto& node_i : mrDestinationModelPart.Nodes())
+        #pragma omp parallel for
+        for(int node_itr=0; node_itr < static_cast<int>(mrDestinationModelPart.NumberOfNodes()); node_itr++)
         {
+            auto node_i = mrDestinationModelPart.NodesBegin() + node_itr;
```

Comment:
```suggestion
            auto node_i = *(mrDestinationModelPart.NodesBegin() + node_itr);
```

Reply:
Btw I would do the following:
~~~c++
const auto nodes_begin = mrDestinationModelPart.NodesBegin();

for ...
~~~